src/test/scala/treadle/DynamicMemorySearch.scala

```diff
@@ -55,130 +56,118 @@ class DynamicMemorySearch extends FreeSpec with Matchers {
 
     val showLocalDebug = false
 
-    val optionsManager = new TreadleOptionsManager {
-      treadleOptions = treadleOptions.copy(
-        writeVCD = false,
-        vcdShowUnderscored = false,
-        setVerbose = false,
-        showFirrtlAtLoad = false,
-        rollbackBuffers = 0,
-        symbolsToWatch = Seq()
-      )
-    }
+    val tester = TreadleTester(Seq(FirrtlSourceAnnotation(input)))
 
-    new TreadleTester(input, optionsManager) {
+    val list: Array[Int] = Array.fill(n)(0)
+    random.setSeed(0L)
 
-      val list: Array[Int] = Array.fill(n)(0)
-      random.setSeed(0L)
+    def startSearch(searchValue: BigInt): Unit = {
+      tester.poke("io_isWr", 0)
+      tester.poke("io_data", searchValue)
+      tester.poke("io_en", 1)
+      tester.step()
+      tester.poke("io_en", 0)
+    }
 
-      def startSearch(searchValue: BigInt): Unit = {
-        poke("io_isWr", 0)
-        poke("io_data", searchValue)
-        poke("io_en", 1)
-        step()
-        poke("io_en", 0)
+    def checkSearch(searchValue: BigInt): Unit = {
+      val expectedIndex = list.indexOf(searchValue) match {
+        case -1 => list.length
+        case x => x
       }
 
-      def checkSearch(searchValue: BigInt): Unit = {
-        val expectedIndex = list.indexOf(searchValue) match {
-          case -1 => list.length
-          case x => x
-        }
-
-        var waitCount = 0
-        while(waitCount <= n && peek("io_done") == Big0) {
-          // println(s"Waiting for done $waitCount")
-          // println(this.engine.circuitState.prettyString())
-          // println(s"external list ${list.mkString(",")}")
-          step()
-          waitCount += 1
-        }
-
-        // println(s"Done wait count is $waitCount done is ${peek("io_done")} " +
-        //   s"got ${peek("io_target")} got $expectedIndex")
-        expect("io_done", 1)
-        expect("io_target", expectedIndex)
+      var waitCount = 0
+      while(waitCount <= n && tester.peek("io_done") == Big0) {
+        // println(s"Waiting for done $waitCount")
+        // println(this.engine.circuitState.prettyString())
+        // println(s"external list ${list.mkString(",")}")
```

Comment:
Nit: I'd prefer to either keep comments out or to convert them to `info` that prints.